packages/sample-app/sample_app/mcp_dev_assistant_server.py

```
@@ -0,0 +1,387 @@
+            modified=modified,
+            untracked=untracked,
+            ahead=ahead,
+            behind=behind
+        )
+        
+    except Exception as e:
+        raise RuntimeError(f"Failed to get Git status: {str(e)}")
+
+
+@server.tool()
+def search_code(pattern: str, directory: str = ".", file_extensions: List[str] = None, max_results: int = 50) -> Dict[str, List[str]]:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)